terraform/transform_orphan.go

```diff
@@ -74,8 +67,27 @@ func (t *OrphanTransformer) Transform(g *Graph) error {
 			state = state.View(t.View)
 		}
 
-		// Go over each resource orphan and add it to the graph.
-		resourceOrphans := state.Orphans(config)
+		var resourceOrphans []string
+		if len(t.Targets) < 1 {
+			// Go over each resource orphan and add it to the graph.
+			resourceOrphans = state.Orphans(config)
+		} else {
+			resourceOrphansExcludingTargets := state.Orphans(config)
+
+			for i, v := range resourceOrphansExcludingTargets {
+				shouldRemove := false
+				for _, t := range t.Targets {
+					prefix := fmt.Sprintf("%s.%s.%d", t.Type, t.Name, t.Index)
+					if !strings.HasPrefix(v, prefix) {
+						shouldRemove = true
+					}
+				}
+				if !shouldRemove {
+					resourceOrphans = append(resourceOrphans, resourceOrphansExcludingTargets[i])
+				}
+			}
+		}
```

Comment:
I think going about this filter in the other direction might be a bit more straightforward. Let's see how it looks:

```go
resourceOrphans := state.Orphans(config)
if len(t.Targets) > 0 {
  var targetedOrphans []string
  for _, o := range resourceOrphans {
    targeted := false
    for _, t := range t.Targets {
      prefix := fmt.Sprintf("%s.%s.%d", t.Type, t.Name, t.Index)
      if strings.HasPrefix(o, prefix) {
        targeted = true
      }
    }
    if targeted {
      targetedOrphans = append(targetedOrphans, o)
    }
  }
  resourceOrphans = targetedOrphans
}
```

Seems sort of nice to have the targeting behavior all behind the guard and the normal case is the straight-through code path. What do you think?